selamlar, 

dediğin gibi yaptım, hata yakalama noktasında biraz daha geliştirilebillir gibi.

kobo için çalıştırdığımda bir sorun yaşamadım. gene de bu testi sadece src klasörünün içinde dictgen betiği varken denediğimi söylemeliyim. stabil bir environment ve test ortamın varsa orada denemen güzel olabilir.
kindle için çalıştırdığımda şu sonucu aldım:

```text
Not running kindlegen, the raw files are located in .../src/GTSv2.4.3_Kindle/GTSv2.4.3_Kindle
Provide KindleGen path with: --write-options 'kindlegen_path=...'
```

sistemimde buna uygun bir executable yok zaten. bu çıktıyı vermesi normal. en azından hata vermiyor. önceden de şu hatayı alıyordum:
```text
Traceback (most recent call last):
  File ".../src/senv/lib/python3.11/site-packages/pyglossary/glossary_v2.py", line 991, in _write
    self._writeEntries(writerList, filename, options)
  File ".../src/senv/lib/python3.11/site-packages/pyglossary/glossary_v2.py", line 937, in _writeEntries
    gen.send(None)
  File ".../src/senv/lib/python3.11/site-packages/pyglossary/plugins/ebook_mobi/writer.py", line 308, in write
    proc = subprocess.Popen(
           ^^^^^^^^^^^^^^^^^
  File ".../.local/share/uv/python/cpython-3.11.13-linux-x86_64-gnu/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File ".../.local/share/uv/python/cpython-3.11.13-linux-x86_64-gnu/lib/python3.11/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'kindlegen'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File ".../src/gts.py", line 692, in <module>
    gts.main()
  File ".../src/gts.py", line 654, in main
    self.kindle()
  File ".../src/gts.py", line 640, in kindle
    glossary.write(filename=str(dosya_ismi), formatName="Mobi", kindlegen_path="kindlegen")
  File ".../src/senv/lib/python3.11/site-packages/pyglossary/glossary_v2.py", line 879, in write
    return self._write(
           ^^^^^^^^^^^^
  File ".../src/senv/lib/python3.11/site-packages/pyglossary/glossary_v2.py", line 995, in _write
    raise WriteError(str(e)) from e
pyglossary.glossary_utils.WriteError: [Errno 2] No such file or directory: 'kindlegen'
Traceback locals:
    self = <pyglossary.glossary_v2.Glossary object at 0x71841b93e4d0>
    filename = '.../src/GTSv2.4.3_Kindle/GTSv2.4.3_...
    len(filename) = 67
    formatName = 'Mobi'
    sort = False
    options = {'kindlegen_path': 'kindlegen'}
    writer = <ebook_mobi.writer.Writer object at 0x718412228690>
    writerList = [<ebook_mobi.writer.Writer object at 0x718412228690>]
```

--- a/src/gts.py
+++ b/src/gts.py
@@ -28,7 +28,7 @@ def which_ex(cmd: str) -> str:
     cwd = Path.cwd()
     arama_dizini = f"{path}{os.pathsep}{cwd}"
     sonuc = which(cmd=cmd, path=arama_dizini)
-    if not sonuc: 
+    if not sonuc:
         return sonuc
     return str(Path(sonuc).resolve())
 
@@ -626,7 +626,7 @@ class GTS:
                 "[!] Kobo biçim dönüşümünü yapacak çalıştırılabilir dosya PATH'de ve betiğin olduğu klasörde bulunamadı. " \
                 "Dosyanın PATH'de veya betiğin çalıştığı klasörde bulunabilir olduğundan emin olun. " \
                 "Dosyaları edinmek için https://github.com/pgaskin/dictutil/releases adresine başvurun.")
-        
+
         subprocess.Popen([dictgen_yolu, str(dosya_ismi),
                           "-o", str(klasor / "dicthtml-tr.zip")],
                           stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
@@ -638,6 +638,11 @@ class GTS:
         if not klasor.exists():
             klasor.mkdir()
         kindlegen_yolu = which_ex("kindlegen")
+        if not kindlegen_yolu:
+            raise Exception(
+                "[!] Kindle MOBI dönüşümünü yapacak çalıştırılabilir dosya (kindlegen) PATH'de ve betiğin olduğu klasörde bulunamadı. " \
+                "Dosyanın PATH'de veya betiğin çalıştığı klasörde bulunabilir olduğundan emin olun. " \
+                "Dosyaları edinmek için https://web.archive.org/web/20190817070956/https://www.amazon.com/gp/feature.html?docId=1000765211 adresine başvurun.")
         glossary.write(filename=str(dosya_ismi), formatName="Mobi", kindlegen_path=kindlegen_yolu)
         mobi_dosya_yolu = dosya_ismi / "OEBPS" / "content.mobi"
         if mobi_dosya_yolu.exists():

